File: Java-OOP-Basics/Problems Course Instance 2016_2017/src/Lesson01DefiningClasses/Exercise/pr08_pokemon_trainer/Pokemon.java
package Lesson01DefiningClasses.Exercise.pr08_pokemon_trainer;

class Pokemon {
    private static final int HEALTH_HIT_ON_LOST_BATTLE = 10;
    private String name;
    private String element;
    private int health;

    Pokemon(String name, String element, int health) {
        super();
        this.name = name;
        this.element = element;
        this.health = health;
    }

    String getElement() {
        return element;
    }

    void loseTournament() {
        this.health -= HEALTH_HIT_ON_LOST_BATTLE;
    }

    boolean isDead() {
        return this.health <= 0;
    }
}

File: C#-OOP-Basics/Problems Course Instance 2018/DefiningClasses/PokemonTrainer/Pokemon.cs
public class Pokemon
{
    public string Name { get; set; }
    public int Health { get; set; }
    public string Element { get; set; }

    public Pokemon() { }
    public Pokemon(string name, int health, string element)
    {
        this.Name = name;
        this.Health = health;
        this.Element = element;
    }
}